### PR TITLE
WIP: Add rule that verifies if OCP4 cluster is upgradeable

### PR DESCRIPTION
**Pull request by JAORMX**

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule ocp_cluster_upgradeable:
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.

File: applications/openshift/ocp_cluster_upgradeable/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            <unix:object object_ref="object_file_for_{{{ rule_id }}}"/>
+    </unix:file_test>
+
+    <unix:file_object id="object_file_for_{{{ rule_id }}}" version="1">
+      <unix:filepath var_ref="{{{ rule_id }}}_dump_location"/>
+    </unix:file_object>
+
+    <ind:yamlfilecontent_object id="object_{{{ rule_id }}}" version="1">
+      <ind:filepath var_ref="{{{ rule_id }}}_dump_location"/>
+      <ind:yamlpath>.status.conditions[type=Upgradeable].status</ind:yamlpath>
```
Comment (mrogers950) [inline applications/openshift/ocp_cluster_upgradeable/oval/shared.xml:39]: if you use `<ind:yamlpath>.status.conditions[:]</ind:yamlpath>` then the regex match can look for `'status: "True", type: Upgradeable'`

Comment (evgenyz) [reply applications/openshift/ocp_cluster_upgradeable/oval/shared.xml:39]: It won't work, the probe would only return scalar values or OVAL sets of scalar values.

**Comment by redhatrises** [comment]
@JAORMX you going to get back to this?

**Comment by redhatrises** [comment]
Closing as the work on this PR has stalled and hasn't been worked on in more than a while.

**Pull request closed without merging**